File: backend/src/main/java/com/otaku/community/feature/feedback/entity/Feedback.java
package com.otaku.community.feature.feedback.entity;

import com.otaku.community.common.entity.BaseEntity;
import com.otaku.community.feature.user.entity.User;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.Index;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "feedbacks", indexes = {
        @Index(name = "idx_feedbacks_reporter_id", columnList = "reporter_id"),
        @Index(name = "idx_feedbacks_status", columnList = "status"),
        @Index(name = "idx_feedbacks_type", columnList = "type")
})
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Feedback extends BaseEntity {

    @Enumerated(EnumType.STRING)
    @Column(name = "type", nullable = false)
    private FeedbackType type;

    @Column(name = "title")
    private String title;

    @Column(name = "content", nullable = false, columnDefinition = "TEXT")
    private String content;

    @Enumerated(EnumType.STRING)
    @Column(name = "target_type")
    private FeedbackTargetType targetType;

    @Column(name = "target_id")
    private String targetId;

    @Enumerated(EnumType.STRING)
    @Column(name = "status", nullable = false)
    private FeedbackStatus status = FeedbackStatus.NEW;

    @Enumerated(EnumType.STRING)
    @Column(name = "priority")
    private FeedbackPriority priority;

    @Enumerated(EnumType.STRING)
    @Column(name = "reason")
    private ReportReason reason;

    @Column(name = "moderator_notes", columnDefinition = "TEXT")
    private String moderatorNotes;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "moderator_id")
    private User moderator;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reporter_id")
    private User reporter;

    @Column(name = "reporter_email")
    private String reporterEmail;

    @Column(name = "reporter_name")
    private String reporterName;

    @Column(name = "is_anonymous", nullable = false)
    private boolean isAnonymous = false;
}

File: frontend/src/lib/api/feedback.ts
import { apiClient } from "./client";
import type { ApiResponse } from "@/types/api";
import type {
  FeedbackType,
  FeedbackTargetType,
  ReportReason,
  FeedbackPriority,
} from "@/types/admin";

export interface FeedbackRequest {
  type: FeedbackType;
  title?: string;
  content: string;
  targetType?: FeedbackTargetType;
  targetId?: string;
  priority?: FeedbackPriority;
  reason?: ReportReason;
  reporterEmail?: string;
  reporterName?: string;
  anonymous?: boolean;
}

export interface FeedbackResponse {
  id: string;
  type: FeedbackType;
  title: string | null;
  content: string;
  status: string;
  createdAt: string;
}

export const feedbackApi = {
  submit: (data: FeedbackRequest) =>
    apiClient.post<ApiResponse<FeedbackResponse>>("/v1/feedbacks", data),
};
